common/primitive-types/src/primitives.rs

```diff
@@ -58,6 +58,10 @@ impl Address {
     pub fn is_zero(&self) -> bool {
         self.0.iter().all(|e| 0_u8.eq(e))
     }
+
+    pub fn as_slice(&self) -> &[u8] {
```

Comment:
is this not `AsRef`?

Reply:
Agreed, `AsRef` should be used wherever possible

Reply:
good point, let's do `AsRef`, changing also #6018 accordingly